### ISSUE TITLE
Add relation rel-during and class history-segment 

### ISSUE THREAD

**Issue by zhengj2007** [BFO-ontology/bfo-trunk#175]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 15, 2013 06:43:48_

to allow for complete use of histories for certain kinds of temporal statements.

rel-during 
domain continuant
range history-segment
inverse (invent name)

a rel-during b =
forall t, t part-of temporal-projection-of b
  a rel (history-of b)

history-segment =
temporal-part-of some history
subclass of history

history-segment-of
domain history-segment
range continuant
functional
inverse has-history-segment

_Original issue: http://code.google.com/p/bfo/issues/detail?id=176_

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 27, 2013 14:30:21_

An example inference, from discussion with Barry, explaining need for history-segment-of to be functional.

We want:

part-of-during o is-history-of < part-of-at-some-time.

in other words, if  a part-of-during b and b is-history-of c then a part-of-at-some-time b

or

s1 segment of history of a and 
s2 segment of history of b
and s1 part of s2
=>
a part-of-at-some-time b

a part-of-during some (history-segment-of b that has-temporal-projection ( is-temporal-projection of s1))

in other words:
s1 segment of history of a and 
s2 segment of history of b
and s1 part of s2
then
exists(h,t)
  a part-of-during h and
  h history-segment-of b and
  h has-temporal-projection t and
  t is-temporal-projection of s1

--

(I think this can be made into a property chain, so we get (effectively) transitive part during some process)

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on June 05, 2013 11:20:35_

Discussed on call. Go with it for now, as long as it doesn't slow release and subject to review.
